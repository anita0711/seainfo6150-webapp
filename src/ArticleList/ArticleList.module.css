.container {
  background: #999;
  border: 1px dashed #666;
  border-radius: 10px;
  box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
  font-family: Arial, Helvetica, sans-serif;
  list-style-type: none;
  padding: 15px;
}

@supports (display: grid) {
  .container {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (min-width: 150px) {
  .container {
    grid-template-columns: repeat(1, 1fr);
    padding: 30px;
    gap: 15px;
    background-color: powderblue;
  }
}

@media screen and (min-width: 500px) and (max-width: 800px) {
  .container {
    grid-template-columns: repeat(2, 1fr);
    padding: 25px;
    gap: 15px;
    background-color: lightpink;
  }
}

@media screen and (min-width: 800px) {
  .container {
    grid-template-columns: repeat(3, 1fr);
    padding: 25px;
    gap: 15px;
    background: #999;
  }
}
