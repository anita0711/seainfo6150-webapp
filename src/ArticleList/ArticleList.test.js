import React from "react";
import ArticleList from "./ArticleList";

const article = [
  {
    "181118211616-banasthali-vidyapith-aviation-school-india": {
      author: "Roxie Parks",
      displayDate: "November 21st 2018, 6:43 pm",
      shortText:
        "On a dusty, 3,600-foot-long airstrip in Rajasthan, India, a two-seater white Cessna 152 pivots into position. A few moments later, it's soaring over the empty desert into a cloudless blue sky.",
      title: "Training India's next generation of female pilots",
    },
  },
];

describe("ArticleList tests", () => {
  it("renders correctly after array is passed", () => {
    const { container } = render(
      <ArticleList articles={Object.values(article)} />
    );
    expect(container).toMatchSnapshot();
  });
});
