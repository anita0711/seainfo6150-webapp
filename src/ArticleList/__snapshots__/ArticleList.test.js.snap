// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly after array is passed 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            />
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
